lib/config/presets/internal/workarounds.ts

```diff
@@ -118,4 +119,24 @@ export const presets: Record<string, Preset> = {
       },
     ],
   },
+  javaLTSVersions: {
+    description: 'Limit Java runtime versions to LTS releases',
+    packageRules: [
+      {
+        description:
+          'Limit Java runtime versions to LTS releases. To receive all major releases add `javaLTSVersions` to the `ignorePresets` array.',
+        matchDatasources: ['docker', 'adoptium-java'],
+        matchPackageNames: [
+          'eclipse-temurin',
+          'amazoncorretto',
+          'adoptopenjdk',
+          'openjdk',
+          'java',
+        ],
+        versioning:
+          'regex:^(?<major>\\d+)?(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?([\\._+](?<build>\\d+))?(-(?<compatibility>.*))?$',
```

Comment:
If you removed this versioning setting, what would be the impact? I think `allowedVersions` would still work, so is this fixing something else?

Reply:
I did try without the versioning regex but it wasn't behaving as expected. In the morning I can create a reproduction and also see if using the suggested non-capturing groups helps

Reply:
Actually that's ok, I think I understand now